pkg/cloud/aws/actuators/machine_scope.go

```diff
@@ -144,25 +150,33 @@ func (m *MachineScope) Close() {
 		m.Error(err, "failed to create new JSONPatch for machine")
 		return
 	}
-	pb, err := json.MarshalIndent(p, "", "  ")
-	if err != nil {
-		m.Error(err, "failed to json marshal patch for machine")
-		return
+
+	if len(p) != 0 {
+		pb, err := json.MarshalIndent(p, "", "  ")
+		if err != nil {
+			m.Error(err, "failed to json marshal patch for machine")
+			return
+		}
+
+		result, err := m.MachineClient.Patch(m.Machine.Name, types.JSONPatchType, pb)
+		if err != nil {
+			m.Error(err, "failed to patch machine")
+			return
+		}
+		// Keep the resource version updated so the status update can succeed
+		m.Machine.ResourceVersion = result.ResourceVersion
 	}
 
-	updated, err := m.MachineClient.Patch(m.Machine.Name, types.JSONPatchType, pb)
-	if err != nil {
-		m.Error(err, "failed to patch machine")
+	// Do not update status if the statuses are the same
+	if reflect.DeepEqual(m.MachineStatus, oldStatus) {
 		return
 	}
 
-	updated.Status.ProviderStatus = status
-	if _, err := m.MachineClient.UpdateStatus(updated); err != nil {
+	m.Machine.Status.ProviderStatus = newStatus
+	if _, err := m.MachineClient.UpdateStatus(m.Machine); err != nil {
 		m.Error(err, "failed to update cluster status")
```

Comment:
Not related to the diff, but this should be: `"failed to update machine status"`